internal/adb/adb_test.go

```
@@ -0,0 +1,728 @@
+		}
+		s.mu.Lock()
+		s.conns = append(s.conns, conn)
+		s.mu.Unlock()
+		go s.handle(conn)
+	}
+}
+
+// handle reads packets and responds; records are pushed before the reply is
+// written, so once the client's call returns its requests are visible.
+func (s *fakeServer) handle(conn net.Conn) {
+	defer conn.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `conn.Close` is not checked

(errcheck)